File: src/main/java/net/hectus/neobb/turn/default_game/block/TDragonHead.java
package net.hectus.neobb.turn.default_game.block;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.SupernaturalClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.block.Block;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class TDragonHead extends BlockTurn implements BuffFunction, SupernaturalClazz {
    public TDragonHead(NeoPlayer player) { super(player); }
    public TDragonHead(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 6;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Effect(Buff.BuffTarget.OPPONENTS, PotionEffectType.BLINDNESS), new Buff.Effect(Buff.BuffTarget.OPPONENTS, PotionEffectType.SLOWNESS), new Buff.Luck(Buff.BuffTarget.OPPONENTS, -20));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/block/TDragonHead.kt
package net.hectus.neobb.modes.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.SupernaturalClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block
import org.bukkit.potion.PotionEffectType

class TDragonHead(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffFunction, SupernaturalClazz {
    override val cost: Int = 6

    override fun buffs(): List<Buff<*>> {
        return listOf(
            Effect(PotionEffectType.BLINDNESS, target = Buff.BuffTarget.OPPONENTS),
            Effect(PotionEffectType.SLOWNESS, target = Buff.BuffTarget.OPPONENTS),
            Luck(-20, Buff.BuffTarget.OPPONENTS)
        )
    }
}
